src/core/variables/ColorVariable.ts

```diff
@@ -53,7 +53,7 @@ export class ColorVariable extends Variable implements IColorVariableParams {
     callback?: IVariableCallback,
   ) {
     super(key, VariableType.COLOR, defaultValue, callback);
-    this.possibleValues = possibleValues;
+    this.possibleValues = possibleValues ? possibleValues : [];
```

Comment:
Unfortunately I don't think we can do this here. `ColorVariable` is implementing the `IColorVariableParams` interface which allows for `possibleValues` to be optional. There is not really a clean way to set an optional property with a `defaultValue` this way. Do you have any suggestion otherwise?